src/utilities/did/did.ts

```diff
@@ -1,54 +1,80 @@
-import {
-  ConfigService,
+import type {
   Did,
   DidDocument,
-  DidEncryptionKey,
-  DidUri,
-} from '@kiltprotocol/sdk-js';
+  KiltKeyringPair,
+  UriFragment,
+  VerificationMethod,
+} from '@kiltprotocol/types';
+
+import { DidResolver } from '@kiltprotocol/sdk-js';
+import {
+  didKeyToVerificationMethod,
+  getFullDid,
+  parse,
+  parseDocumentFromLightDid,
+} from '@kiltprotocol/did';
+
+import { blake2AsHex } from '@polkadot/util-crypto';
 
 import { useAsyncValue } from '../useAsyncValue/useAsyncValue';
 
-export function isFullDid(did: DidUri | undefined): boolean {
+export function isFullDid(did: Did | undefined): boolean {
   if (!did) {
     // maybe the DID was deleted from the blockchain
     return false;
   }
-  return Did.parse(did).type === 'full';
+  return parse(did).type === 'full';
 }
 
-export async function getFullDidDocument(did: DidUri): Promise<DidDocument> {
-  const api = ConfigService.get('api');
-  return Did.linkedInfoFromChain(await api.call.did.query(Did.toChain(did)))
-    .document;
+export async function getFullDidDocument(did: Did): Promise<DidDocument> {
+  const { didDocument } = await DidResolver.resolve(did, {});
+  if (!didDocument) {
+    throw new Error('Unable to resolve DID');
+  }
+  return didDocument;
 }
 
-export function useFullDidDocument(did: DidUri): DidDocument | undefined {
+export function useFullDidDocument(did: Did): DidDocument | undefined {
   return useAsyncValue(getFullDidDocument, [did]);
 }
 
-export async function getDidDocument(did: DidUri): Promise<DidDocument> {
+export async function getDidDocument(did: Did): Promise<DidDocument> {
   return isFullDid(did)
     ? await getFullDidDocument(did)
-    : Did.parseDocumentFromLightDid(did);
+    : parseDocumentFromLightDid(did);
 }
 
 export function getDidEncryptionKey({
   keyAgreement,
-}: DidDocument): DidEncryptionKey {
+}: DidDocument): UriFragment {
   if (!keyAgreement || !keyAgreement[0]) {
     throw new Error('encryptionKey is not defined somehow');
   }
   return keyAgreement[0];
 }
 
-export function parseDidUri(did: DidUri): ReturnType<typeof Did.parse> & {
-  fullDid: DidUri;
+export function parseDidUri(did: Did): ReturnType<typeof parse> & {
+  fullDid: Did;
 } {
-  const parsed = Did.parse(did);
-  const fullDid = parsed.type === 'full' ? did : Did.getFullDidUri(did);
+  const parsed = parse(did);
+  const fullDid = parsed.type === 'full' ? did : getFullDid(did);
 
   return {
     ...parsed,
     fullDid,
   };
 }
+
+export function computeKeyId(key: Uint8Array): VerificationMethod['id'] {
+  return `#${blake2AsHex(key, 256)}`;
+}
```

Comment:
If the SDK people are asking us to attach this gun to our foot, we should really push back. How come application code has to know KILT internals in such depth?

Reply:
In that case the key ID has to be computed beforehand so it can be passed as a signer to the batch transaction.

Reply:
That’s an unfortunate outcome of the design

Reply:
This shouldn't be necessary. The result of you adding an assertionMethod in the same transaction in which you're using it is that you'll have to pass a 'doctored' DID document to the helper that selects the right signer with which to authorise  the call - the function relies on the mapping between purposes and verification methods in the DID document. Because you're messing with the document you can use whatever id you choose, as long as it corresponds to the id on the signer for the key pair you're using here. So may as well be `#assert` or something